full-moon/src/ast/types.rs

```diff
@@ -302,3 +302,31 @@ impl<'a> TypeSpecifier<'a> {
         &self.type_info
     }
 }
+
+/// A compound assignment operator, such as `+=` or `-=`
+#[derive(Clone, Debug, Display, PartialEq, Owned, Node, Visit)]
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+#[display(fmt = "{}{}", lhs, rhs)]
+pub struct CompoundAssignment<'a> {
```

Comment:
This needs to contain the `TokenReference` to the compound assignment operator, otherwise we have no idea what it is.